rclcpp_examples/CMakeLists.txt

```diff
@@ -101,23 +101,35 @@ if(AMENT_ENABLE_TESTING)
     find_package("${middleware_impl}" REQUIRED)
   endforeach()
 
-  foreach(example_executable list_parameters_async list_parameters parameter_events_async parameter_events set_and_get_parameters_async set_and_get_parameters)
+  # Add each test case.  Multi-executable tests can be specified in
+  # colon-separated strings, like  exe1:exe2.
+  foreach(example_executable list_parameters_async list_parameters parameter_events_async parameter_events set_and_get_parameters_async set_and_get_parameters talker:listener)
+
+    set(RCLCPP_EXAMPLES_EXPECTED_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/test/${example_executable}")
+    string(
+      REPLACE ":" ":${CMAKE_CURRENT_SOURCE_DIR}/test/"
+      RCLCPP_EXAMPLES_EXPECTED_OUTPUT ${RCLCPP_EXAMPLES_EXPECTED_OUTPUT})
```

Comment:
Instead of this custom logic with a colon separator and then applying the same prefix twice (L108 and L109-111) I would suggest using something more "native". Simply making each element a CMake list:

    foreach(example_executable list_parameters_async list_parameters parameter_events_async parameter_events set_and_get_parameters_async set_and_get_parameters "talker;listener")

And then apply the same operation on each item of the list. That would avoid duplication of logic and help to make the code more readable.

Same below.